### ISSUE TITLE
FieldImage cannot be navigated to when opt_clickHandler is passed

### ISSUE THREAD

**Issue by samelhusseini** [google/blockly#3339]
Currently, field click is determined whether or not the block is editable and the showEditor_ method is implemented.
Keyboard Navigation is determined whether or not the field is EDITABLE. This means that FieldImage that has an optional click handler passed in, cannot be navigated to.

Perhaps FieldImage should set EDITABLE to true, but then it would be SERIALIZABLE due to this code: https://github.com/google/blockly/blob/master/core/field.js#L434 that doesn't let fields be EDITABLE and SERIALIZABLE at the same time.

If we don't see a reason for a field to be EDITABLE and not SERIALIZABLE then we should at least fix keyboard nav to use isFieldClickable instead.

@alschmiedt @BeksOmega thoughts?

**Comment by BeksOmega** [comment]
EDITABLE and SERIALIZABLE are my least favorite parts of the fields API, just saying that hehe.

The only thing EDITABLE is currently used for is determining if the block should have different UI depending on if the block `isEditable` or not.

The reason it's not used to tell anything about editors is because I thought it would be kind of silly for people to define showEditor_ but not have it work unless they also set EDITABLE to true.

It seems like the best option for keyboard nav would be to use `field.isClickable()` (and maybe rename it to something else like `isInteractable`).  I don't know much about keyboard nav though, so I could be wrong.

**Comment by samelhusseini** [comment]
Yeah I agree, it makes sense for keyboard nav to use the same reasoning for whether or not to allow interacting as gesture.js.

**Comment by RoboErikG** [comment]
Side note, keyboard nav should be able to navigate to labels too so they can be read out by screen readers. Some cursor implementations may want to skip them, but it shouldn't be the default.

**Comment by alschmiedt** [comment]
I can make a feature request for this. Just to clarify though navigating to the labels as is will not allow them to be read out by the screen reader. There is more work that would need to be done before the screen reader and keyboard navigation play nicely.

**Comment by rachel-fenichel** [comment]
@alschmiedt is this done?
